File: app/src/main/java/ds/features/model/TimeProvider.java
package ds.features.model;

import android.databinding.BaseObservable;
import android.databinding.Bindable;
import ds.features.BR;

import java.text.SimpleDateFormat;
import java.util.Date;

public class TimeProvider extends BaseObservable {

	Date d;


	public TimeProvider() {
		d = new Date();
	}


	@Bindable
	public String getTime() {
		//d.setTime(System.currentTimeMillis());
		return new SimpleDateFormat("hh:mm:ss").format(d);
	}

	public void setTime(long time){
		d.setTime(time);
		notifyPropertyChanged(BR.time);
	}
}

File: app/src/main/java/ds/features/model/TimeProvider.kt
package ds.features.model

import android.databinding.BaseObservable
import android.databinding.Bindable
import ds.features.BR
import java.text.SimpleDateFormat
import java.util.*

class TimeProvider : BaseObservable() {

	internal var d = Date()

	val time: String
		@Bindable
		get() = SimpleDateFormat("hh:mm:ss").format(d)

	fun setTime(time: Long) {
		d.time = time
		notifyPropertyChanged(BR.time)
	}
}
